### PR TITLE
Code cleanup for mod/item.php

### PR DESCRIPTION
**Pull request by annando**
This is some kind of intermediate PR. This is a preparation to make mod/item.php being able to be moved to some Module class.

### CODE REVIEW

**Review by MrPetovan — APPROVED** [review]

**Comment by nupplaphil** [comment]
The phpunit is currently stuck because my server crashed during an RAM upgrade with all Friendica and CI instances on it ;-) .. *gg*

**Review by MrPetovan — APPROVED** [review]

File: mod/item.php [hunk 1]
```diff
@@ -38,47 +37,30 @@
...
 use Friendica\Protocol\Activity;
-use Friendica\Security\Security;
 use Friendica\Util\DateTimeFormat;
-use Friendica\Util\ParseUrl;
 
 function item_post(App $a) {
-	if (!DI::userSession()->isAuthenticated()) {
+	$uid = DI::userSession()->getLocalUserId();
+
+	if (!DI::userSession()->isAuthenticated() || !$uid) {
```
Comment (MrPetovan) [inline mod/item.php]: You only need the $uid part as the “isAuthenticated” method will alway be true if the $uid is set.
```suggestion
	if (!$uid) {
```

File: mod/item.php [hunk 2]
```diff
@@ -94,30 +76,97 @@ function item_post(App $a) {
...
+
+	$json = ['success' => 1];
+	System::jsonExit($json);
+}
+
+function item_edit(int $uid, array $request, bool $preview, string $return_path)
+{
+	$post = Post::selectFirst(Item::ITEM_FIELDLIST, ['id' => $request['post_id'], 'uid' => $uid]);
+	if (!DBA::isResult($post)) {
+		DI::sysmsg()->addNotice(DI::l10n()->t('Unable to locate original post.'));
```
Comment (MrPetovan) [inline mod/item.php]: This is superfluous if we throw an Exception. Please move this inside the $return_path block.

File: mod/item.php [hunk 3]
```diff
@@ -94,30 +76,97 @@ function item_post(App $a) {
...
+	$post['pubmail']   = $request['pubmail_enable'] ?? false;
+	$post['created']   = $request['created_at'] ?? DateTimeFormat::utcNow();
+	$post['edited']    = $post['changed'] = $post['commented'] = $post['created'];
+	$post['app']       = '';
+	$post['inform']    = '';
+	$post['postopts']  = '';
+	$post['file']      = '';
+
+	if ($post['parent']) {
+		if ($post['parent']) {
```
Comment (MrPetovan) [inline mod/item.php]: Duplicate condition, you can remove it.

File: mod/item.php [hunk 4]
```diff
@@ -94,30 +76,97 @@ function item_post(App $a) {
...
+			$parent_item = Post::selectFirst(Item::ITEM_FIELDLIST, ['id' => $post['parent']]);
 		}
 
 		// if this isn't the top-level parent of the conversation, find it
 		if (DBA::isResult($parent_item)) {
```
Comment (MrPetovan) [inline mod/item.php]: DBA::isResult is superfluous with “selectFirst”
```suggestion
		if ($parent_item) {
```

File: mod/item.php [hunk 5]
```diff
@@ -136,620 +185,154 @@ function item_post(App $a) {
...
-				0 => [
-					'src'    => $attachment_img_src,
-					'width'  => $attachment_img_width,
-					'height' => $attachment_img_height
-				]
-			];
-		} else {
-			unset($attachment['images']);
+	$post = Post::selectFirst(Item::ITEM_FIELDLIST, ['id' => $post_id]);
+	if (!DBA::isResult($post)) {
```
Comment (MrPetovan) [inline mod/item.php]: Same here
```suggestion
	if (!$post) {
```

File: mod/item.php [hunk 6]
```diff
@@ -136,620 +185,154 @@ function item_post(App $a) {
...
-			$attachment = Attach::selectFirst($fields, ['id' => $mtch]);
-			if ($attachment !== false) {
-				if (strlen($attachments)) {
-					$attachments .= ',';
-				}
-				$attachments .= Post\Media::getAttachElement(DI::baseUrl() . '/attach/' . $attachment['id'],
-					$attachment['filesize'], $attachment['filetype'], $attachment['filename'] ?? '');
-			}
-			$body = str_replace($match[1],'',$body);
+		DI::sysmsg()->addNotice(DI::l10n()->t('Empty post discarded.'));
```
Comment (MrPetovan) [inline mod/item.php]: Please move this inside the return_path block

File: src/Content/Item.php [hunk 7]
```diff
@@ -756,4 +788,237 @@ public function addShareLink(string $body, int $quote_uri_id): string
...
+			$post['allow_cid'] = $toplevel_item['allow_cid'] ?? '';
+			$post['allow_gid'] = $toplevel_item['allow_gid'] ?? '';
+			$post['deny_cid']  = $toplevel_item['deny_cid'] ?? '';
+			$post['deny_gid']  = $toplevel_item['deny_gid'] ?? '';
+			$post['private']   = $toplevel_item['private'];
+			return $post;
+		}
+
+		$user = User::getById($post['uid'], ['allow_cid', 'allow_gid', 'deny_cid', 'deny_gid']);
+		if (!DBA::isResult($user)) {
```
Comment (MrPetovan) [inline src/Content/Item.php]:
```suggestion
		if (!$user) {
```

File: src/Content/Item.php [hunk 8]
```diff
@@ -756,4 +788,237 @@ public function addShareLink(string $body, int $quote_uri_id): string
...
+			// The ACL selector introduced in version 2019.12 sends ACL input data even when the Public visibility is selected
+			$post['allow_cid'] = $post['allow_gid'] = $post['deny_cid'] = $post['deny_gid'] = '';
+		} else if ($visibility === 'custom') {
+			// Since we know from the visibility parameter the item should be private, we have to prevent the empty ACL
+			// case that would make it public. So we always append the author's contact id to the allowed contacts.
+			// See https://github.com/friendica/friendica/issues/9672
+			$post['allow_cid'] .= $this->aclFormatter->toString(Contact::getPublicIdByUserId($post['uid']));
+		}
+
+		if (strlen($post['allow_gid']) || strlen($post['allow_cid']) || strlen($post['deny_gid']) || strlen($post['deny_cid'])) {
```
Comment (MrPetovan) [inline src/Content/Item.php]: Superfluous function calls.
```suggestion
		if ($post['allow_gid'] || $post['allow_cid'] || $post['deny_gid'] || $post['deny_cid']) {
```

File: src/Content/Item.php [hunk 9]
```diff
@@ -756,4 +788,237 @@ public function addShareLink(string $body, int $quote_uri_id): string
...
+		if (!preg_match_all('/(\[attachment\]([0-9]+)\[\/attachment\])/',$post['body'], $match)) {
+			return $post;
+		}
+
+		foreach ($match[2] as $attachment_id) {
+			$attachment = Attach::selectFirst(['id', 'uid', 'filename', 'filesize', 'filetype'], ['id' => $attachment_id, 'uid' => $post['uid']]);
+			if (empty($attachment)) {
+				continue;
+			}
+			if (strlen($post['attach'])) {
```
Comment (MrPetovan) [inline src/Content/Item.php]:
```suggestion
			if ($post['attach']) {
```

File: src/Content/Item.php [hunk 10]
```diff
@@ -756,4 +788,237 @@ public function addShareLink(string $body, int $quote_uri_id): string
...
+			Tag::createImplicitMentions($post['uri-id'], $post['thr-parent-id']);
+		}
+
+		Hook::callAll('post_local_end', $post);
+
+		$author = DBA::selectFirst('contact', ['thumb'], ['uid' => $post['uid'], 'self' => true]);
+
+		foreach ($recipients as $recipient) {
+			$address = trim($recipient);
+			if (!strlen($address)) {
```
Comment (MrPetovan) [inline src/Content/Item.php]:
```suggestion
			if (!$address) {
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: mod/item.php (hunk 1 above)
Comment (annando) [reply mod/item.php]: Done

File: mod/item.php (hunk 2 above)
Comment (annando) [reply mod/item.php]: Done

File: src/Content/Item.php (hunk 10 above)
Comment (annando) [reply src/Content/Item.php]: Done

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**